src/collisions_updater.cpp

```diff
@@ -0,0 +1,228 @@
+
+        return true;
+    }
+
+    moveit_setup_assistant::LinkPairMap compute(uint32_t trials, double min_collision_fraction,  bool verbose){
+        unsigned int collision_progress;
+        return moveit_setup_assistant::computeDefaultCollisions(config_data.getPlanningScene(),
+                                                                &collision_progress,
+                                                                trials > 0,
+                                                                trials,
+                                                                min_collision_fraction, verbose);
+    }
+
+    void write(const moveit_setup_assistant::LinkPairMap &link_pairs, bool include_default, bool include_always, const std::string &output_path){
+        // Create temp disabled collision
+        srdf::Model::DisabledCollision dc;
+
+        std::set<SortableDisabledCollision> disabled_collisions;
+        disabled_collisions.insert(config_data.srdf_->disabled_collisions_.begin(), config_data.srdf_->disabled_collisions_.end());
+
+        // copy the data in this class's LinkPairMap datastructure to srdf::Model::DisabledCollision format
+        for ( moveit_setup_assistant::LinkPairMap::const_iterator pair_it = link_pairs.begin();
+                pair_it != link_pairs.end(); ++pair_it)
+        {
+            // Only copy those that are actually disabled
+            if( pair_it->second.disable_check ){
+                if( pair_it->second.reason == moveit_setup_assistant::DEFAULT && !include_default) continue;
+                if( pair_it->second.reason == moveit_setup_assistant::ALWAYS && !include_always) continue;
+                dc.link1_ = pair_it->first.first;
+                dc.link2_ = pair_it->first.second;
+                dc.reason_ = moveit_setup_assistant::disabledReasonToString( pair_it->second.reason );
+                disabled_collisions.insert(SortableDisabledCollision(dc));
+            }
+        }
+
+        config_data.srdf_->disabled_collisions_.assign(disabled_collisions.begin(), disabled_collisions.end());
+
+        // Update collision_matrix for robot pose's use
+        config_data.loadAllowedCollisionMatrix();
+
+        config_data.srdf_->writeSRDF(output_path.empty() ? config_data.srdf_path_ : output_path );
+
+    }
+
+};
+
+int main(int argc, char * argv[]){
+
+    std::string config_pkg_path;
+    std::string urdf_path;
+    std::string srdf_path;
+
+    std::string output_path;
+
+    bool include_default = false, include_always = false, keep_old = false, verbose = false;
+
+    double min_collision_fraction = 1.0;
+
+    uint32_t never_trials = 0;
+
+    po::options_description desc("Allowed options");
+    desc.add_options()
+        ("help", "show help")
+        ("config-pkg", po::value(&config_pkg_path), "path to moveit config package")
+        ("urdf", po::value(&urdf_path), "path to URDF ( or xacro)")
+        ("srdf", po::value(&srdf_path), "path to SRDF ( or xacro)")
```

Comment:
Ah I see, you've added support for this here in ``loadFileToString()`` but there is not support for it in the main GUI version. I think there is some duplicate functionality between ``start_screen_widget.cpp`` and this executable that would better be merged. This way the main GUI also can support SRDFs in xacro format. A much cleaner implementation